sunpy/map/mapbase.py

```diff
@@ -613,13 +613,12 @@ def wcs(self):
         w2.wcs.cunit = self.spatial_units
         w2.wcs.aux.rsun_ref = self.rsun_meters.to_value(u.m)
 
-        w2.wcs.dateavg = self.reference_date.isot
-        if self._date_obs is not None:
-            w2.wcs.dateobs = self._date_obs.isot
+        w2.wcs.dateobs = self.date.utc.isot
```

Comment:
What's the reasoning for including date-obs on the wcs if it hasn't come from the date-obs header?

Reply:
Our metadata approach/model, where a source subclass is supposed to fix a wrong DATE-OBS or redirect to a non-standard keyword by overriding `.date`.  For constructing the WCS object, we should look only at (potentially overridden) properties, not at the keywords themselves.  (Technically `_date_obs` is a property that could be overridden, but it is private and undocumented.)  So, the use cases basically fall into three categories (far from an equipartition):

1. The user wants the canonical observation time on the WCS object and DATE-OBS is correct.
2. The user wants the canonical observation time on the WCS object and the `.date` property has been overridden to not pull it straight from DATE-OBS.
3. The user doesn't care whether the WCS object has the canonical observation time.

Sticking `.date` in `.wcs.wcs.dateobs` in all cases means that (1) and (2) are satisfied and (3) has not been negatively impacted.